### PR TITLE
Update rubocop 0.56.0 -> 1.12.1

### PR DESCRIPTION
**Pull request by tokuhirom**
- Disable NewCops by default
- Follow the rubycop's incompatible changes(e.g. rule name modification)
- Apply automatic correct for lib/line/bot/client.rb
- Do not use RUBY_VERSION in line-bot-api.gemspec
  https://sue445.hatenablog.com/entry/2016/09/02/121047
- Install rubycop-rake, rubycop-rspec
  (suggested by rubocop)

### CODE REVIEW

File: line-bot-api.gemspec [hunk 1]
```diff
@@ -20,7 +20,6 @@ Gem::Specification.new do |spec|
   spec.required_ruby_version     = '>= 2.4.0'
 
   spec.add_development_dependency "addressable", "~> 2.3"
-  spec.add_development_dependency "bundler", "~> 1.11" if RUBY_VERSION < "2.3"
```
Comment (tokuhirom) [inline line-bot-api.gemspec:23]: Do not use RUBY_VERSION in line-bot-api.gemspec
https://sue445.hatenablog.com/entry/2016/09/02/121047

File: lib/line/bot/client.rb [hunk 2]
```diff
@@ -954,7 +954,7 @@ def parse_events_from(request_body)
       def validate_signature(content, channel_signature)
         return false if !channel_signature || !channel_secret
 
-        hash = OpenSSL::HMAC.digest(OpenSSL::Digest::SHA256.new, channel_secret, content)
```
Comment (tokuhirom) [inline lib/line/bot/client.rb:957]: OpenSSL::Digest::SHA256 is a deprecated constant. This is fixed by rubocop.

File: lib/line/bot/client.rb [hunk 3]
```diff
@@ -986,6 +986,7 @@ def content_type(file)
         if file.respond_to?(:content_type)
           content_type = file.content_type
           raise ArgumentError, "invalid content type: #{content_type}" unless ['image/jpeg', 'image/png'].include?(content_type)
+
```
Comment (tokuhirom) [inline lib/line/bot/client.rb:989]: rubocop added this new line

File: Gemfile [hunk 4]
```diff
@@ -4,6 +4,8 @@ gemspec
 
 group :development, :test do
   # ref: http://docs.rubocop.org/en/latest/installation/
-  gem 'rubocop', '~> 0.56.0', require: false
+  gem 'rubocop', '~> 1.12.1', require: false
+  gem 'rubocop-rake', '~> 0.3.1', require: false
+  gem 'rubocop-rspec', '~> 2.4.0', require: false
```
Comment (tokuhirom) [inline Gemfile:9]: rubocop suggest to install these plugins

File: .rubocop.yml [hunk 5]
```diff
@@ -34,32 +35,40 @@ Style/EachWithObject:
...
+Layout/LineLength:
+  Max: 200
 Layout/SpaceInsideBlockBraces:
   Enabled: false
+Layout/HashAlignment:
+  Enabled: false
 Layout/SpaceInsideHashLiteralBraces:
   Enabled: false
-Layout/IndentHeredoc:
+Layout/HeredocIndentation:
```
Comment (tokuhirom) [inline .rubocop.yml:71]: This rule was renamed

File: .rubocop.yml [hunk 6]
```diff
@@ -34,32 +35,40 @@ Style/EachWithObject:
...
-Metrics/LineLength:
-  Max: 200
 Metrics/BlockLength:
   Enabled: false
 Metrics/MethodLength:
   Enabled: false
 Metrics/ClassLength:
   Enabled: false
 
+Layout/LineLength:
```
Comment (tokuhirom) [inline .rubocop.yml:63]: Ranemd from Metrics/LineLength

File: .rubocop.yml [hunk 7]
```diff
@@ -34,32 +35,40 @@ Style/EachWithObject:
...
+  Max: 200
 Layout/SpaceInsideBlockBraces:
   Enabled: false
+Layout/HashAlignment:
+  Enabled: false
```
Comment (tokuhirom) [inline .rubocop.yml:68]: This rule is not compatible with the current code base. I disabled this rule.

File: .rubocop.yml [hunk 8]
```diff
@@ -34,32 +35,40 @@ Style/EachWithObject:
...
+Naming/MethodParameterName:
   Enabled: false
 Naming/ConstantName:
   Enabled: false
 Naming/AccessorMethodName:
   Enabled: false
 Naming/MethodName:
   Enabled: false
+Naming/FileName:
+  Enabled: false
```
Comment (tokuhirom) [inline .rubocop.yml:54]: This rule suggest to rename `lib/line-bot-sdk.rb` to CamelCase. We can't accept the suggestion.

File: .rubocop.yml [hunk 9]
```diff
@@ -34,32 +35,40 @@ Style/EachWithObject:
...
   Enabled: false
+Style/StringConcatenation:
+  Enabled: false
+Style/RedundantAssignment:
+  Enabled: false
 
 Naming/HeredocDelimiterNaming:
   Enabled: false
-Naming/UncommunicativeMethodParamName:
+Naming/MethodParameterName:
```
Comment (tokuhirom) [inline .rubocop.yml:45]: This rule was renamed

File: .rubocop.yml [hunk 10]
```diff
@@ -34,32 +35,40 @@ Style/EachWithObject:
   Enabled: false
 Style/FrozenStringLiteralComment:
   Enabled: false
+Style/StringConcatenation:
```
Comment (tokuhirom) [inline .rubocop.yml:38]: The following 2 rules are not compatible with the current code base.

File: .rubocop.yml [hunk 11]
```diff
@@ -2,6 +2,7 @@ AllCops:
   TargetRubyVersion: 2.4
   Exclude:
     - 'examples/**/*'
+  NewCops: disable
```
Comment (tokuhirom) [inline .rubocop.yml:5]: Do not enable new rules by default.

**Review by toduq — APPROVED** [review]

**Review by kimoto — APPROVED** [review]

**Pull request merged**